service/history/replication/executable_history_task.go

```diff
@@ -435,10 +442,14 @@ func (e *ExecutableHistoryTask) checkEvents(incomingEvents []*historypb.HistoryE
 	if currentNewRunEvents != nil {
 		return serviceerror.NewInvalidArgument("Current Task is expected to be the last event of a workflow")
 	}
-	currentLastEvent := currentEvents[len(currentEvents)-1]
-	incomingFirstEvent := incomingEvents[0]
+
+	currentLastBatch := currentEvents[len(currentEvents)-1]
+	currentLastEvent := currentLastBatch[len(currentLastBatch)-1]
+	incomingFirstBatch := incomingEventBatches[0]
+	incomingFirstEvent := incomingFirstBatch[0]
+
 	if currentLastEvent.Version != incomingFirstEvent.Version {
-		return serviceerror.NewInvalidArgument("events version mismatch")
+		return serviceerror.NewInvalidArgument("events version does not match")
```

Comment:
Is it worth reporting the version numbers here?

Reply:
Not for now. We can add it in the future if it is necessary.